universe/engine.py

```diff
@@ -1,110 +1,101 @@
-from decimal import Decimal
-import random
+from . import components, systems
+from .orders import (Move, CargoTransfer, Scrap, BuildInstallation, Terraform,
+                     BuildStation, BuildShip, LaunchMassPacket)
 
 
-class GameState(object):
+class Entity:
+    def __init__(self, registry, data):
+        self._data = data
+        self._components = registry[data['type']]
+
+    def __contains__(self, key):
+        return key in self._components
+
+    def serialize(self):
+        data = {'type': self._data['type']}
```

Comment:
Thoughts: I think type should probably be provided by a mandatory `Metadata` component.  Iterating over the Metadata collection, then, would be a good way of getting all entities down in `Manager.export_data`, rather than unioning over all available component collections.